resteasy-client-reactor-netty/src/main/java/org/jboss/resteasy/client/jaxrs/engines/ReactorNettyClientHttpEngine.java

```diff
@@ -196,6 +170,81 @@ public ReactorNettyClientHttpEngine(final HttpClient httpClient,
                 });
     }
 
+    /**
+     * The main business logic mapping RestEasy's {@link ClientInvocation request} to Reactor Netty's concept
+     * of it.
+     */
+    private HttpClient.ResponseReceiver<?> send(final ClientInvocation resteasyReq) {
+        final Optional<Object> reqPayload = Optional.ofNullable(resteasyReq.getEntity());
+
+        final HttpClient.RequestSender requestSender = httpClient.headers(headers -> addHeaders(resteasyReq, headers))
+                    .request(HttpMethod.valueOf(resteasyReq.getMethod()))
+                    .uri(resteasyReq.getUri().toString());
+
+        return reqPayload.<HttpClient.ResponseReceiver<?>>map(ignore -> {
+            return requestSender.send((reactorReq, outbound) -> {
+                final ByteBufOutputStream byteBufOutputStream =
+                    new ByteBufOutputStream(outbound.alloc().buffer());
+
+                /* Replacing the ClientRequestHeaders with TrackingClientRequestHeaders
+                to track the changes to headers by the sendRequestBody method. */
+                resteasyReq.setHeaders(
+                    new TrackingClientRequestHeaders(
+                        resteasyReq.getClientConfiguration(),
+                        resteasyReq.getHeaders().getHeaders()
+                    )
+                );
+
+                try {
+                    sendRequestBody(resteasyReq, byteBufOutputStream);
+                } catch (final IOException ioe) {
+                    return Mono.error(ioe);
```

Comment:
@anilgursel `sendRequestBody` was originally wrapping `IOException` with `RuntimeException`.  Do you see any problem with the replacement I've done here?

Reply:
You could use an `UncheckedIOException` which extends a `RuntimeException` but is more descriptive.

Reply:
I'd like to stick with `Mono#error` over over using a nicer unchecked exception unless any of our reactor gurus complain.  I think this is considered kosher and won't have any unnecessary wrapping done solely to deal with lambda+checked-exceptions.

That said, I did not know that `UncheckedIOException` existed.  Thanks for the heads up!

Reply:
Ah I see. I thought you meant before it was doing something like `Mono.error(new RuntimeException(ioe))`. Yes, using `Mono.error` is definitely better than throwing an explicit exception.

Reply:
@crankydillo I cannot think of a problem at this moment.